include/sparrow/arrow_array_schema_proxy.hpp

```diff
@@ -374,14 +377,68 @@ namespace sparrow
          */
         [[nodiscard]] SPARROW_API arrow_proxy view() const;
 
+        /**
+         * Check if the proxy owns the `ArrowArray` and not only a pointer to it.
+         */
         [[nodiscard]] SPARROW_API bool owns_array() const;
```

Comment:
I think we should not expose the implementation detail in the documentation. What about something like:
`Check whether the proxy has ownership of its internal the ArrowArray` 
?

Reply:
ok